runtime/lua/vim/diagnostic.lua

```diff
@@ -1073,6 +1073,7 @@ function M.show_line_diagnostics(opts, bufnr, lnum)
   opts = opts or {}
   opts.focus_id = "line_diagnostics"
   opts.lnum = lnum or (vim.api.nvim_win_get_cursor(0)[1] - 1)
+  bufnr = get_bufnr(bufnr)
   local line_diagnostics = M.get(bufnr, opts)
```

Comment:
reassigning a param in the middle of a function body is a bit icky, maybe inlining the expression is better?
```suggestion
  local line_diagnostics = M.get(get_bufnr(bufnr), opts)
```

Reply:
It may be icky, but it's the pattern used throughout the file so it's a familiar pattern for readers.